### PR TITLE
[New Rule] Out-Of-Tree Kernel Module Load

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This rule monitors the syslog log file for messages related to instances of a out-of-tree kernel module load, indicating the taining of the kernel. Rootkits often leverage kernel modules as their main defense evasion technique. Detecting tainted kernel module loads is crucial for ensuring system security and integrity, as malicious or unauthorized modules can compromise the kernel and lead to system vulnerabilities or unauthorized access.

## Detection
```
host.os.type:linux and event.dataset:"system.syslog" and process.name:kernel and 
message:"loading out-of-tree module taints kernel."
```
image

### CODE REVIEW

File: rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- To run the system module of Filebeat on Linux follow the setup instructions in the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-system.html).
+
+"""
+severity = "low"
+tags = [
+        "Domain: Endpoint",
+        "OS: Linux",
+        "Use Case: Threat Detection",
+        "Tactic: Persistence",
+        "Tactic: Defense Evasion"
```
Comment (Aegrah) [inline rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml:54]:
```suggestion
        "Tactic: Defense Evasion",
        "Rule Type: BBR"
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+can compromise the kernel and lead to system vulnerabilities or unauthorized access.
+"""
+from = "now-9m"
+index = ["logs-system.syslog-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Tainted Out-Of-Tree Kernel Module Load"
+risk_score = 21
+rule_id = "51a09737-80f7-4551-a3be-dac8ef5d181a"
+setup = """
```
Comment (w0rk3r) [inline rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml:26]:
```suggestion
setup = """## Setup
```

File: rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+- To run Filebeat on Docker follow the setup instructions in the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/current/running-on-docker.html).
+- To run Filebeat on Kubernetes follow the setup instructions in the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/current/running-on-kubernetes.html).
+- For quick start information for Filebeat refer to the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/8.11/filebeat-installation-configuration.html).
+- For complete Setup and Run Filebeat information refer to the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/current/setting-up-and-running.html).
+
+#### Rule Specific Setup Note
+- This rule requires the Filebeat System Module to be enabled.
+- The system module collects and parses logs created by the system logging service of common Unix/Linux based distributions.
+- To run the system module of Filebeat on Linux follow the setup instructions in the [helper guide](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-system.html).
+
```
Comment (w0rk3r) [inline rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml:46]:
```suggestion
```

File: rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+[rule.threat.tactic]
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+
+[[rule.threat.technique]]
+name = "Rootkit"
+id = "T1014"
+reference = "https://attack.mitre.org/techniques/T1014/"
```
Comment (w0rk3r) [inline rules_building_block/persistence_tainted_kernel_module_out_of_tree_load.toml:93]:
```suggestion
[[rule.threat.technique]]
name = "Rootkit"
id = "T1014"
reference = "https://attack.mitre.org/techniques/T1014/"

[rule.threat.tactic]
id = "TA0005"
name = "Defense Evasion"
reference = "https://attack.mitre.org/tactics/TA0005/"
```
nit

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**